cupy/functional/piecewise.py

```diff
@@ -0,0 +1,73 @@
+import cupy
+
+from cupy import core
+
+_piecewise_krnl = core.ElementwiseKernel(
+    'U condlist, raw S funclist, int64 xsiz',
+    'raw I prev, raw T y',
+    '''
+        if (condlist && prev[i % xsiz] < i)
+            prev[i % xsiz] = i;
+        __syncthreads();
+        if (prev[i % xsiz] == i)
+            y[i % xsiz] = funclist[i / xsiz];
+        ''',
+    'piecewise_kernel'
+)
+
+
+def piecewise(x, condlist, funclist):
+    """Evaluate a piecewise-defined function.
+
+        Args:
+            x: input domain
```

Comment:
Write the type of arguments.
```suggestion
            x (cupy.ndarray): input domain
```